### PR TITLE
Add-on store sort by new

### LINKED ISSUE #16681: Add ability to view new available add-ons
### Is your feature request related to a problem? Please describe.
Users may want to be notified when new add-ons are available from the store.
### Describe the solution you'd like
- A New add-ons tab maybe added to the store, where last compatible add-ons not cached would be presented.
- A dialog or other means, like an ui.message or a toast notification, would notify users when new add-ons are available, for example at startup or at scheduled intervals (each hour may be reasonable.
- The dialog will be the store focusing the New add-ons tab, where the description and other details, an an action to install add-ons, would be available.
### Describe alternatives you've considered
An add-on maybe created to notify about new add-ons, but it maybe better to include this in the core adding a new tab to the store.
### Additional context
A bot used on Mastodon, on a Feeds subgroup of the add-ons mailing list and on a spanish mailing list is used by some people to be notified about new add-ons:

- [Bot repo](https://github.com/nvdaes/nvdaStoreMastodonBot)
- [Bot profile on Mastodon](https://mastodon.chrisduffley.com/@nvdastore)

### PR DESCRIPTION
**Pull request by nvdaes**
Add ability to view new add-ons

### Link to issue number:
Fixes #16681
### Summary of the issue:
NVDA should have the ability to present new add-ons first.
### Description of user facing changes
- The add-on list can have a new column with the publication date of add-ons.
- The publication date can be shown in the add-ons details panel.
- Add-ons can be sorted by date (from more recent to older and from older to recent), using the sort by column feature.

### Description of development approach
- Functions have been changed in add-on store gui, and propertys have been added in addon.py module of the addonStore.

### Testing strategy:
Tested manually, adding a submissionTime key to the json file of various installed add-ons.

### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [ ] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Added a submission time attribute for add-ons to improve tracking and management.
	- Introduced a publication date property for better usability in displaying add-on information.
	- Added a combo box in settings to allow users to configure the order of add-ons displayed in the store.

- **Improvements**
	- Enhanced sorting functionality for add-ons based on user-defined preferences.
	- Updated user guide to include new sorting options for better user understanding.
	- Provided clear instructions for sorting options and metadata management in the add-on store documentation.

### CODE REVIEW

**Comment by nvdaes** [comment]
@coderabbitai review

**Comment by nvdaes** [comment]
@coderabbitai, I have changed the issue title addressing your comment.

**Comment by nvdaes** [comment]
@coderabbitai, can you suggest how to document manual testing for this?

**Comment by nvdaes** [comment]
@coderabbitai, I'm a blind person and I'd like to know if the size for the publication date column (50) is wider enough.

**Comment by nvdaes** [comment]
@coderabbitai review

**Comment by nvdaes** [comment]
@coderabbitai, do  you think that this maybe improved in some way?

**Comment by nvdaes** [comment]
@coderabbitai, I've fixed the grammar issue. Anything else that I can do?

**Comment by seanbudd** [comment]
>In contrast, the publication date can be formatted, but sorting items by publication date can produce different results, which may not match timestamp order.
From https://github.com/nvaccess/nvda/issues/16681#issuecomment-2323166796

@nvdaes - you should be able to solve this problem by creating a custom [`wx.ListCtrl`](https://docs.wxpython.org/wx.ListCtrl.htm) subclass using:
- `SortItems` to create a custom sort function
- `GetSortIndicator` to get the current column index for `SortItems`
- `SetItemData` to set the date used for sorting
- `ShowSortIndicator` to make this indicator more obvious

**Comment by nvdaes** [comment]
> > In contrast, the publication date can be formatted, but sorting items by publication date can produce different results, which may not match timestamp order.
> > From [#16681 (comment)](https://github.com/nvaccess/nvda/issues/16681#issuecomment-2323166796)
> 
> @nvdaes - you should be able to solve this problem by creating a custom [`wx.ListCtrl`](https://docs.wxpython.org/wx.ListCtrl.htm) subclass using:
> 
>     * `SortItems` to create a custom sort function
> 
>     * `GetSortIndicator` to get the current column index for `SortItems`
> 
>     * `SetItemData` to set the date used for sorting
> 
>     * `ShowSortIndicator` to make this indicator more obvious

OK Sean. @coderabbitai, can you provide usage samples for SortItems function?

File: source/config/configSpec.py [hunk 1]
```diff
@@ -335,6 +335,7 @@
 [addonStore]
 	showWarning = boolean(default=true)
 	automaticUpdates = option("notify", "disabled", default="notify")
+	addonListOrder = option("recent", "older", "alphabetical", default="recent")
```
[the remark this answers is not in the document]
Comment (nvdaes) [reply source/config/configSpec.py:338]: OK, I'll try to do it. For now I'll try to sort items by publication date by default, from recent to older, until further comments.

**Comment by nvdaes** [comment]
@coderabbitai, can you suggest a way to use SetItemData after a ctrlList has been created?

**Comment by nvdaes** [comment]
@coderabbitai, how to get the items of a previously created ctrlList? Is there any method for this?

**Comment by nvdaes** [comment]
I'm working on this, but I don't know how to make the publication date header column to appear at the top of the add-ons list to click it.
Ifsomeone can assist with this, I'll be grateful.

**Comment by CyrilleB79** [comment]
Sorry @nvdaes, could you be more specific? What you are trying to achieve is not clear to me.

**Comment by nvdaes** [comment]
Cyrille wrote:

> could you be more specific? What you are trying to achieve is not clear to me.

I'mtrying to use the procedure explained by you in issue #15277 to sort by columns. But when I switch to screen review, I don't find the header corresponding to the pubication date,even if this is presented for items with a submissionTime.

**Comment by nvdaes** [comment]
@CyrilleB79 , thanks for your reply. Seems that I have fixed the mentioned issue where the date header wasn't shown, though I don't know if my approach is visually correct.
But at least this will be useful for completing the work and then a visual person may provide more advice.
I'll explain how to sort add-ons by columns in the user guide, as requested by @seanbudd , and my explanation will be based on your comment about this.

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ In order to use this feature, the application volume adjuster needs to be enable
 * Added an action in the Add-on Store to cancel the install of add-ons. (#15578, @hwf1324)
 * Added an action in the Add-on Store to retry the installation if the download/installation of an add-on fails. (#17090, @hwf1324)
 * It is now possible to specify a mirror URL to use for the Add-on Store. (#14974)
-* The add-ons lists in the Add-on Store can be sorted by columns. (#15277, @nvdaes)
+* The add-ons lists in the Add-on Store can be sorted by columns, included publication date, in ascending and descending order. (#15277, #16681, @nvdaes)
```
Comment (seanbudd) [inline user_docs/en/changes.md:18]:
```suggestion
* The add-ons lists in the Add-on Store can be sorted by columns, including publication date, in ascending and descending order. (#15277, #16681, @nvdaes)
```

**Review by seanbudd — APPROVED** [review]
Looks good to me, including the new column widths. Thanks for your work @nvdaes

**Pull request merged**